libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -242,9 +259,21 @@ bool AP_Camera::update_location(const struct Location &loc)
         // be called without a new GPS fix
         return false;
     }
+
     if (get_distance(loc, _last_location) < _trigg_dist) {
         return false;
     }
-    _last_location = loc;
-    return true;
+
+    if ((unsigned) ahrs.roll_sensor/100.0f > _max_roll) {
```

Comment:
Unless I'm missing something with what ahrs.roll_sensor reports don't you need to take the absolute vaue? Otherwise I would imagine extreme rolls to the opposite direction wouldn't be prevented.